.github/workflows/main.yaml

```diff
@@ -261,16 +261,24 @@ jobs:
           target: ${{ matrix.job.target }}
 
   build_envoy:
-    runs-on: neco-containers-ubuntu-20.04
+    runs-on: neco-containers-ubuntu-22.04
     steps:
       - uses: actions/checkout@v4
+      - name: Read TAG
+        run: |
+          TAG=$(cat ./envoy/TAG)
+          echo "TAG=${TAG}" >> $GITHUB_ENV
+          VERSION=$(echo $TAG | cut -d \. -f 1,2,3)
+          echo "VERSION=${VERSION}" >> $GITHUB_ENV
       - name: Build container image
         uses: ./.github/actions/build_envoy
         with:
-          version: 1.28.1
+          version: ${{ env.VERSION }}
           clang_archive_path: https://github.com/llvm/llvm-project/releases/download/llvmorg-14.0.0/clang+llvm-14.0.0-x86_64-linux-gnu-ubuntu-18.04.tar.xz
           github_token: ${{ secrets.GITHUB_TOKEN }}
           yamory_token: ${{ secrets.YAMORY_ACCESS_TOKEN }}
+      - name: Test envoy
```

Comment:
I would like to move all the container-specific logic to each build job to keep `main.yaml` cleaner.